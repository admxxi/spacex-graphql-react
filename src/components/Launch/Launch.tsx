import React from 'react'
import styled, { withTheme } from 'styled-components'
import { Launch as ILaunch } from '../../@types/graphql'
import { mySlugify } from '../../helpers/index'
import Link from 'next/link'
interface LaunchProps {
  launch: ILaunch
}

const Launch = styled(({ launch }: LaunchProps) => {
  return (
    <div className="card">
      <div className="card-content">
        <div className="media">
          <div className="media-content">
            <p className="title is-4">
              {launch.id}
              {launch.details}
            </p>
            <Link
              href={{
                pathname: '/mission/[slug]',
                query: {
                  slug: `${mySlugify(String(launch.id), true)}`
                }
              }}
            >
              <a>{launch.details}</a>
            </Link>
          </div>
        </div>
      </div>
    </div>
  )
})``

export default withTheme(Launch)
