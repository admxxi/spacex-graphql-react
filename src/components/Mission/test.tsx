import React from 'react'
import renderer from 'react-test-renderer'
import Launch from './Launch'

it('renders correctly', () => {
  const tree = renderer.create(<Launch />).toJSON()
  expect(tree).toMatchSnapshot()
})
