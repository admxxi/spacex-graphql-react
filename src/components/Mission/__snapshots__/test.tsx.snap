// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="Mission__Style-wzd66q-0 lhGnqF"
>
  <article
    class="card"
  >
    <div
      class="card-content"
    >
      <div
        class="media is-align-content-center"
      >
        <div
          class="media-content"
        >
          <a
            class="is-link is-3"
          >
            Crew-1
          </a>
          <p
            class="subtitle is-4 mt-2"
          >
            SpaceX will launch the first operational mission of its Crew Dragon vehicle as part of NASA's Commercial Crew Transpo...
          </p>
          <div
            class="ma-1 subtitle is-5 is-italic has-text-right mx-1"
          >
            11/16/2020
          </div>
        </div>
      </div>
    </div>
  </article>
</div>
`;
